controllers/helloworld_controller.go

```diff
@@ -124,62 +168,43 @@ func (r *HelloworldReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 	}
 
-	helloworld.Status.AvailableReplicas = int32(currentReplicas)
+	helloworld.Status.AvailableReplicas = int32(len(podList.Items))
 	helloworld.Status.ReadyReplicas = int32(countReady)
 
 	Logging(req, "Updating status")
 
 	if err := r.Status().Update(ctx, helloworld); err != nil {
 		log.Log.Error(err, "unable to update Helloworld")
-		return ctrl.Result{}, err
-	}
-
-	Logging(req, "Finished succesfully")
-
-	return ctrl.Result{Requeue: false}, nil
-}
-
-func (r *HelloworldReconciler) svcForHelloworld(h *v1alpha1.Helloworld) *v1.Service {
-	lbls := labelsForApp(h.Name)
-
-	svc := &v1.Service{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      h.Name + "-service",
-			Namespace: h.Namespace,
-			Labels:    lbls,
-		},
-		Spec: v1.ServiceSpec{
-			Ports: []v1.ServicePort{
-				{
-					Protocol:   v1.ProtocolTCP,
-					Port:       8080,
-					TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: 8080},
-				}},
-			Selector: lbls,
-			Type:     v1.ServiceTypeClusterIP,
-		},
+		return stopReconcile, ctrl.Result{}, err
 	}
 
-	controllerutil.SetControllerReference(h, svc, r.Scheme)
-	return svc
+	return !stopReconcile, ctrl.Result{}, nil
 }
 
-func (r *HelloworldReconciler) podForHelloworld(h *v1alpha1.Helloworld) *v1.Pod {
-	lbls := labelsForApp(h.Name)
-	podName := h.Name + "-pod-" + fmt.Sprint(time.Now().UnixNano())
+func (r *HelloworldReconciler) definePodFor(helloworld *v1alpha1.Helloworld) *v1.Pod {
+	lbls := labelsForApp(helloworld.Name)
+	podName := helloworld.Name + "-pod-" + fmt.Sprint(time.Now().UnixNano())
+	const svcImage = "gcr.io/run-ai-lab/dans-service"
```

Comment:
the consts should be defined at the top of the page, under the imports.
"""
const (
        svcImage = "gcr.io/run-ai-lab/dans-service"
)
"""